src/main/java/com/puppycrawl/tools/checkstyle/doclets/TokenTypesDoclet.java

```diff
@@ -19,148 +19,288 @@
+            throw new CheckstyleException(message.getMessage());
         }
-        return onlyOneDestFileOption && foundDestFileOption;
+        return status.getTree();
     }
 
     /**
-     * Reads destination file name.
-     * @param options all specified options.
-     * @return destination file name
+     * Converts inline code tag into HTML form.
+     * @param builder to append
+     * @param inlineTag to format
+     * @throws CheckstyleException if the inline javadoc tag is not a code tag
      */
-    private static String getDestFileName(String[]... options) {
-        String fileName = null;
-        for (final String[] opt : options) {
-            if (DEST_FILE_OPT.equals(opt[0])) {
-                fileName = opt[1];
+    private static void formatInlineCodeTag(StringBuilder builder, DetailNode inlineTag)
+            throws CheckstyleException {
+        if (JavadocUtils.findFirstToken(inlineTag, JavadocTokenTypes.CODE_LITERAL) == null) {
+            throw new CheckstyleException("Expected inline @code tag");
+        }
+        builder.append("<code>");
+        for (DetailNode node : inlineTag.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.TEXT) {
+                builder.append(node.getText());
             }
         }
-        return fileName;
+        builder.append("</code>");
+    }
+
+    /**
+     *  Prints the usage information.
+     */
+    private static void printUsage() {
+        final HelpFormatter formatter = new HelpFormatter();
+        formatter.setWidth(HELP_WIDTH);
+        formatter.printHelp(String.format("java %s [options] <input file>.",
+            TokenTypesDoclet.class.getName()), buildOptions());
+    }
+
+    /**
+     * Parses doclet command line based on passed arguments.
+     * @param args
+     *        command line arguments
+     * @return parsed information about passed arguments
+     * @throws ParseException
+     *         when passed arguments are not valid
+     */
+    private static CommandLine parseCli(String... args)
+            throws ParseException {
+        final CommandLineParser clp = new DefaultParser();
+        return clp.parse(buildOptions(), args);
+    }
+
+    /**
+     * Builds and returns the list of parameters supported by doclet cli.
+     * @return available options
+     */
+    private static Options buildOptions() {
+        final Options options = new Options();
+        options.addRequiredOption(null, OPTION_DEST_FILE, true, "The output file.");
+        return options;
     }
+
+    /**
+     * Parses Java source file. Results in AST which contains comment nodes.
+     * @param file file to parse
+     * @return DetailAST tree
+     * @throws NullPointerException if the text is {@code null}
+     * @throws IOException          if the file could not be read
+     * @throws ANTLRException       if parser or lexer failed
+     */
+    private static DetailAST parseFile(String file) throws IOException, ANTLRException {
```

Comment:
This will result in a _horizontal_ dependency. It's worse than just copying a couple of lines.

The right solution, IMHO, is to extract this code into an utility class. May be to the TreeWalker itself. I'll take a close look to it.

Reply:
Done.